tests/test_live.py

```diff
@@ -0,0 +1,488 @@
+        time.sleep(2)
+
+
+def subscriber_receive_second(sub: persipubsub.subscriber.Subscriber) -> None:
+    global FIRST_SUB_STARTED_READING  # pylint: disable=global-statement
+
+    start = time.time()
+    while not FIRST_SUB_STARTED_READING:
+        if time.time() - start > 10:
+            raise TimeoutError
+        time.sleep(0.1)
+
+    with sub.receive() as _:
+        if time.time() - start < 2:
+            assert isinstance(sub.queue, persipubsub.queue._Queue)  # pylint: disable=protected-access
+            assert isinstance(sub.queue.env, lmdb.Environment)
+            assert isinstance(sub.queue.env.path(), str)
+            assert isinstance(sub.identifier, str)
+            result = pathlib.Path(sub.queue.env.path()) / sub.identifier
+            result.write_text('pass')
+        else:
+            assert isinstance(sub.queue, persipubsub.queue._Queue)  # pylint: disable=protected-access
+            assert isinstance(sub.queue.env, lmdb.Environment)
+            assert isinstance(sub.queue.env.path(), str)
+            assert isinstance(sub.identifier, str)
+            result = pathlib.Path(sub.queue.env.path()) / sub.identifier
+            result.write_text('fail')
+
+
+def receive(sub: persipubsub.subscriber.Subscriber,
+            num_msg: int,
+            timeout: int = 2,
+            sleep_time: float = 0,
+            method_timeout: int = 60) -> None:
+    received_msg = 0
+
+    start = time.time()
+    while True:
+        if received_msg == num_msg:
+            break
+        elif time.time() - start >= method_timeout:
+            raise TimeoutError
+
+        with sub.receive(timeout=timeout) as msg:
+            if msg is not None:
+                received_msg += 1
+                time.sleep(sleep_time)
+
+    assert isinstance(sub.queue, persipubsub.queue._Queue)  # pylint: disable=protected-access
+    assert isinstance(sub.queue.env, lmdb.Environment)
+    assert isinstance(sub.queue.env.path(), str)
+    assert isinstance(sub.identifier, str)
+    result = pathlib.Path(sub.queue.env.path()) / sub.identifier
+    result.write_text('pass')
+
+
+def send(pub: persipubsub.publisher.Publisher,
+         num_msg: int,
+         sleep_time: float = 0) -> None:
+    for _ in range(num_msg):
+        pub.send(msg="hello subscriber".encode('utf-8'))
+        time.sleep(sleep_time)
+
+
+class TestLive(unittest.TestCase):
+    def test_multithreaded_communication_one_publisher_one_subscriber(
+            self) -> None:
+        with temppathlib.TemporaryDirectory() as tmp_dir:
+            env = persipubsub.environment.new_environment(path=tmp_dir.path)
+            _ = env.new_control(['sub'])
+
+            result = tmp_dir.path / "sub"
+            result.touch()
+
+            pub = env.new_publisher()
+            sub = env.new_subscriber(identifier='sub')
+
+            num_msg = 1000
+            pub_thread = threading.Thread(
+                target=send, kwargs={
```

Comment:
Would it be possible to not use kwargs? maybe just `sub=sub, num_msg=num_msg` would work?

Reply:
If it is possible to change it, please change it everywhere in the unit tests

Reply:
I looked into it and threading.Threads doesn't seem to accept any method arguments compared to Threadingpool which does. You can only define target=method, args=(..., ..., ...), kwargs={'...': ..., '....': ...}. I could create a wrapper method that accept kwargs in non Mapping[str, Any] form and then transform it to a dict. What do you think?